### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Seems strange but didn't have any warnings or anything... So let's see with travis.

### CODE REVIEW

**Comment by EPashkin** [comment]
If was for `cargo clippy`

**Comment by GuillaumeGomez** [comment]
Ah we don't run it on travis? I thought we did... Fixing warnings then.

**Comment by EPashkin** [comment]
Clippy still don't shipped with rust (but planned) and still need nightly, so we can't add it to stables.

**Comment by GuillaumeGomez** [comment]
Oh that's not an issue. I added it to some of my projects which test both stable and nightly (I'm thinking about sysinfo and process-viewer). I just test clippy and nightly and don't call on stable. Nothing magical but very convenient.

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/lib.rs [hunk 1]
```diff
@@ -144,6 +144,11 @@
 //! them once. **Omitting them in the following cargo invocations will not undo
 //! their effects!**
 
+#![cfg_attr(feature = "cargo-clippy", allow(new_without_default))]
+#![cfg_attr(feature = "cargo-clippy", allow(type_complexity))]
+#![cfg_attr(feature = "cargo-clippy", allow(borrowed_box))]
```
Comment (EPashkin) [inline src/lib.rs]: We fixed this warning for generated code, please fix it in manual too.

**Comment by EPashkin** [comment]
@GuillaumeGomez, interesting idea, maybe really apply it on gtk too?

**Comment by GuillaumeGomez** [comment]
Sure, I'll add it to this PR.

**Comment by GuillaumeGomez** [comment]
And here you go. We now have cargo clippy check on nightly.

**Comment by EPashkin** [comment]
@GuillaumeGomez Thanks, 👍 with changing PR title.

**Pull request merged**